content/blogs/gui-setup-part2-partitioning.md

```diff
@@ -0,0 +1,88 @@
+
+Only once these steps have been performed, the installer inquires the user for the ReactOS installation directory, and how to install the bootloader.
+After that, the ReactOS installation proper begins, with file copy and saving settings, before restarting and continuing the remaining of the installation with the 2nd-stage installation.
+
+
+# Workflow change in the GUI-mode setup
+
+While the installation workflow used by the text-mode USETUP _might_ make sense, from a UI perspective (each action is done from a different screen; one cannot go back to previous screens; and once an action has been selected, it cannot be undone), the GUI-mode setup changes some of these assumptions.
+The wizard-style of the GUI setup allows to go back and forward in the different selection pages.
+Its partitioning page shows a minimal interface, similar to the text-mode one, but more reminiscent of other GUI partitioning software.
+
+Changing the disk partition layout and formatting new or existing partitions can be considered to be a sensitive operation -- existing partitions or filesystems can be destroyed after all!
+The user then may feel more comfortable performing such operations by viewing the resulting changes: new partition layout, etc., knowing they haven't yet been actually applied, until his/her final choice has been made.
+This is why, for example, 3rd-party partitioning software allow to stash (queue) all these changes, allowing for a careful review before actually applying them.
+
+Because of these considerations, I chose to make the ReactOS GUI setup follow a similar philosophy.
+The partition page allows the user to choose the partition where to install ReactOS, as well as to modify the partition layout, keeping the changes in memory until the actual ReactOS installation is started.
+
+<<< IMAGE ROS GUI SETUP PARTITION >>>
+
+This new installation workflow necessitated to adapt the existing partitioning code, that is shared with the text-mode setup; to determine where and how to store additional per-partition arbitrary data state (for example, filesystem choice and formatting parameters).
+
+Most importantly, this required two main changes in the code.
+- Rewriting the formatter/chkdsk machine-state code, that was used in USETUP, to make it more generic and act as a queue whose stashed actions can be then committed at once.
+- splitting the bootloader installation choice, from its actual installation code.
+This was required, because the old behaviour was to present the user with the bootloader installation choice only at the very end of ReactOS installation, and not before starting it, and the old code was not reusable elsewhere.
```

Comment:
```suggestion
  - This was required, because the old behaviour was to present the user with the bootloader installation choice only at the very end of ReactOS installation, and not before starting it, and the old code was not reusable elsewhere.
```